internal/policygen/iam.go

```diff
@@ -154,6 +220,39 @@ func validate(instance map[string]interface{}, mandatoryFields []string) error {
 	return nil
 }
 
+// validateMandatoryLists checks the presence of mandatory fields and assert []interface{} type.
+func validateMandatoryStringLists(instance map[string]interface{}, mandatoryFields []string) error {
+	for _, k := range mandatoryFields {
+		if _, ok := instance[k]; !ok {
+			return fmt.Errorf("mandatory field %q missing from instance: %v", k, instance)
+		}
+		if _, ok := instance[k].([]interface{}); !ok {
+			return fmt.Errorf("value for %q should be a []interface{}, got %T", k, instance[k])
+		}
+		for _, s := range instance[k].([]interface{}) {
+			if _, ok := s.(string); !ok {
+				return fmt.Errorf("%q should be a string, got %T", s, s)
+			}
+		}
+
+	}
+	return nil
+}
+
+func normalizeID(rn runner.Runner, t, id string) (string, error) {
+	var err error
+	// For projects, the ID in the state is the project ID, but we need project number in policies.
+	if t == "project" {
+		if id, err = projectNumber(rn, id); err != nil {
```

Comment:
I think it's unexpected to reuse the input arg as the output arg like this. I would instead just make the two output args named and use those. Something like

```go
func normalizeID(rn runner.Runner, t, id string) (nid string, err error) {
	// For projects, the ID in the state is the project ID, but we need project number in policies.
	if t == "project" {
		if nid, err = projectNumber(rn, id); err != nil {
			return "", err
		}
	} else if t == "folder" {
		// For folders, the ID can be either {folder_id} or folders/{folder_id}. Remove the 'folders/' prefix if exists.
		nid = strings.TrimPrefix(id, "folders/")
	}
	return nid, nil
}
```